tests/BUILD

```diff
@@ -23,6 +23,8 @@ filegroup(
         "@llvm-project//mlir:libmlir_runner_utils.so",
         "@llvm-project//mlir:mlir-cpu-runner",
         "@llvm-project//mlir:mlir-opt",
+        "@edu_berkeley_abc//:abc",
```

Comment:
Sooo agree - one thing I was thinking of was that if we can skip registration based on an env var in case the yosys deps need to be skipped - this goes back to this earlier thing: https://github.com/google/heir/pull/191#discussion_r1364256334

I think I'll go ahead and implement that - I'll add some documentation saying that if you'd like to skip the Yosys build, add this env var and we'll build with a selective configuration.